tests/contract/test_resource_client.py

```diff
@@ -449,57 +449,61 @@ def test_generate_update_example_create_override(resource_client):
     assert example == {"a": 5, "b": 2}
 
 
-def test_has_writable_identifier_primary_is_read_only(resource_client):
+def test_is_primary_identifier_create_only_primary_is_read_only(resource_client):
     resource_client._update_schema(
         {
             "primaryIdentifier": ["/properties/foo"],
             "readOnlyProperties": ["/properties/foo"],
         }
     )
 
-    assert not resource_client.has_writable_identifier()
+    assert not resource_client.is_primary_identifier_create_only()
 
 
-def test_has_writable_identifier_primary_is_writeable(resource_client):
+def test_is_primary_identifier_create_only_primary_is_create_only(resource_client):
     resource_client._update_schema({"primaryIdentifier": ["/properties/foo"]})
 
-    assert resource_client.has_writable_identifier()
+    assert resource_client.is_primary_identifier_create_only()
 
 
-def test_has_writable_identifier_primary_and_additional_are_read_only(resource_client):
+def test_is_primary_identifier_create_only_composite_primary_are_read_only(
+    resource_client,
+):
     resource_client._update_schema(
         {
-            "primaryIdentifier": ["/properties/foo"],
-            "additionalIdentifiers": [["/properties/bar"]],
+            "primaryIdentifier": ["/properties/foo", "/properties/bar"],
             "readOnlyProperties": ["/properties/foo", "/properties/bar"],
         }
     )
 
-    assert not resource_client.has_writable_identifier()
+    assert not resource_client.is_primary_identifier_create_only()
 
 
-def test_has_writable_identifier_additional_is_writeable(resource_client):
+def test_is_primary_identifier_create_only_composite_primary_is_read_only(
+    resource_client,
+):
     resource_client._update_schema(
         {
-            "primaryIdentifier": ["/properties/foo"],
-            "additionalIdentifiers": [["/properties/bar"]],
+            "primaryIdentifier": ["/properties/foo", "/properties/bar"],
             "readOnlyProperties": ["/properties/foo"],
+            "createOnlyProperties": ["/properties/bar"],
         }
     )
 
-    assert resource_client.has_writable_identifier()
+    assert not resource_client.is_primary_identifier_create_only()
 
 
-def test_has_writable_identifier_compound_is_writeable(resource_client):
+def test_is_primary_identifier_create_only_composite_primary_is_create_only(
+    resource_client,
+):
     resource_client._update_schema(
         {
-            "primaryIdentifier": ["/properties/foo"],
-            "additionalIdentifiers": [["/properties/bar", "/properties/baz"]],
-            "readOnlyProperties": ["/properties/foo", "/properties/baz"],
+            "primaryIdentifier": ["/properties/foo", "/properties/bar"],
+            "createOnlyProperties": ["/properties/bar", "/properties/foo"],
```

Comment:
lets not remove additional identifiers just change the results to match what we expect

Reply:
Since the test is only updated primary identifiers and testing them, it maybe confusing to a reader why the additionalIdentifiers are being updated.